src/main/java/seedu/sudohr/model/employee/UniqueEmployeeList.java

```diff
@@ -1,134 +1,281 @@
+     * {@code persons} must not contain duplicate persons.
      */
-    public void setEmployees(List<Employee> employees) {
-        requireAllNonNull(employees);
-        if (!employeesAreUnique(employees)) {
+    public void setEmployees(List<Employee> persons) {
+        requireAllNonNull(persons);
+        if (!personsAreUnique(persons)) {
             throw new DuplicateEmployeeException();
         }
+        if (!phoneNumbersAreUnique(persons)) {
+            throw new DuplicatePhoneNumberException();
+        }
+        if (!emailsAreUnique(persons)) {
+            throw new DuplicateEmailException();
+        }
 
-        internalList.setAll(employees);
+        internalList.setAll(persons);
     }
-
     /**
      * Returns the backing list as an unmodifiable {@code ObservableList}.
      */
     public ObservableList<Employee> asUnmodifiableObservableList() {
         return internalUnmodifiableList;
     }
-
     @Override
     public Iterator<Employee> iterator() {
         return internalList.iterator();
     }
 
+    /**
+     * Returns if two lists are equal by checking the contents of the lists.
+     * @param other the list to check against
+     * @return a boolean value indicating equality.
+     */
     @Override
     public boolean equals(Object other) {
-        return other == this // short circuit if same object
-                || (other instanceof UniqueEmployeeList // instanceof handles nulls
-                        && internalList.equals(((UniqueEmployeeList) other).internalList));
-    }
+        // short circuit if same object
+        if (other == this) {
+            return true;
+        }
+        if (other instanceof UniqueEmployeeList) { // handles null
+            // idea it to make sure each list is a subset of the other
+            Map<Id, Employee> currentList = new HashMap<>();
+            Map<Id, Employee> otherList = new HashMap<>();
+
+            for (Employee e : internalList) {
+                currentList.put(e.getId(), e);
+            }
 
+            for (Employee e : (UniqueEmployeeList) other) {
+                otherList.put(e.getId(), e);
+            }
+
+            for (Id id : currentList.keySet()) {
+                Employee employeeFromCurrent = currentList.get(id);
+                Employee employeeFromOther = otherList.getOrDefault(id, null);
+                if (employeeFromOther == null
+                        || !employeeFromCurrent.equals(employeeFromOther)) {
+                    return false;
+                }
+            }
+
+            for (Id id : otherList.keySet()) {
+                Employee employeeFromCurrent = currentList.getOrDefault(id, null);
+                Employee employeeFromOther = otherList.get(id);
+                if (employeeFromCurrent == null
+                        || !employeeFromCurrent.equals(employeeFromOther)) {
+                    return false;
+                }
+            }
+        }
+        return true;
```

Comment:
I think this is quite a clever check, but if the other object that is being compared is not an instance of UniqueEmployeeList, i think it will return true? Did you mean to put the `return true` inside the `if (same instance) {...} ` block and default is return false?